src/libs/sfdk/dockercontainer.cpp

```diff
@@ -0,0 +1,341 @@
+    arguments.append(QLatin1String(PRIVILEGED2));
+    arguments.append(QLatin1String(VOLUME));
+    arguments.append(QLatin1String(CGROUP_MOUNT));
+    arguments.append(QLatin1String(TMPFS));
+    arguments.append(QLatin1String(TMPFS1));
+    arguments.append(QLatin1String(TMPFS));
+    arguments.append(QLatin1String(TMPFS2));
+    arguments.append(QLatin1String(TMPFS));
+    arguments.append(QLatin1String(TMPFS3));
+
+    for (auto& ports : m_reservedPortMappings) {
+        arguments.append(QLatin1String(PUBLISH_PORT));
+        arguments.append(QString::number(ports.first) + ":" + QString::number(ports.second));
+    }
+
+    for (auto& paths : m_sharedPaths) {
+        arguments.append(QLatin1String(VOLUME));
+        arguments.append(paths.first + ":" + paths.second);
+    }
+
+    arguments.append(q->name());
+
+    auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+    QObject::connect(runner.get(), &ProcessRunner::done, q, [q, process = runner->process()](bool ok) {
+        if (ok)
+            q->m_containerId = QString::fromLocal8Bit(process->readAllStandardOutput().trimmed());
+    });
+    QObject::connect(runner.get(), &ProcessRunner::done, context, functor);
+    SdkPrivate::commandQueue()->enqueue(std::move(runner));
+}
+
+void DockerContainerPrivate::stop(const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    const QPointer<const QObject> context_{context};
+
+    auto enqueue = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+           auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+           QObject::connect(runner.get(), &ProcessRunner::failure, Sdk::instance(), [=]() {
+               SdkPrivate::commandQueue()->cancelBatch(batch);
+               callIf(context_, functor, false);
+           });
+           ProcessRunner *runner_ = runner.get();
+           SdkPrivate::commandQueue()->enqueue(std::move(runner));
+           return runner_;
+       };
+
+    CommandQueue::BatchId batch = SdkPrivate::commandQueue()->beginBatch();
+
+    QStringList arguments;
+    arguments.append(QLatin1String(STOP));
+    arguments.append(q->m_containerId);
+
+    enqueue(arguments, batch);
+
+    arguments[0] = QLatin1String(COMMIT);
+    arguments.append(q->name());
+
+    enqueue(arguments, batch);
+
+    SdkPrivate::commandQueue()->enqueueCheckPoint(context,  [=]() { functor(true); });
+    SdkPrivate::commandQueue()->endBatch();
+}
+
+void DockerContainerPrivate::probe(const QObject *context, const Functor<VirtualMachinePrivate::BasicState, bool> &functor) const
+{
+    Q_Q(const DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    DockerManager::fetchRegisteredVirtualMachines(q,
+        [=](const QStringList &registeredVms, bool ok) {
+            if (!ok) {
+                VirtualMachinePrivate::BasicState state;
```

Comment:
It would be more understandable if defined just once at outer block scope. From that I would understand that this is the outcome that is to be computed.

Reply:
fixed